lake/segment/writer.go

```diff
@@ -1,76 +1,67 @@
 package segment
 
 import (
+	"bytes"
 	"context"
 	"io"
 
+	"github.com/brimdata/zed/field"
 	"github.com/brimdata/zed/lake/seekindex"
 	"github.com/brimdata/zed/order"
 	"github.com/brimdata/zed/pkg/bufwriter"
-	"github.com/brimdata/zed/pkg/nano"
 	"github.com/brimdata/zed/pkg/storage"
-	"github.com/brimdata/zed/zio"
 	"github.com/brimdata/zed/zio/zngio"
 	"github.com/brimdata/zed/zng"
 )
 
 // Writer is a zbuf.Writer that writes a stream of sorted records into a
 // data segment.
 type Writer struct {
-	ref           *Reference
-	byteCounter   *writeCounter
-	count         uint64
-	size          int64
-	rowObject     *zngio.Writer
-	firstTs       nano.Ts
-	lastTs        nano.Ts
-	needSeekWrite bool
-	order         order.Which
-	seekIndex     *seekindex.Builder
-	wroteFirst    bool
+	ref             *Reference
+	byteCounter     *writeCounter
+	count           uint64
+	rowObject       *zngio.Writer
+	firstKey        zng.Value
+	lastKey         zng.Value
+	needSeekWrite   bool
+	order           order.Which
+	seekIndex       *seekindex.Writer
+	seekIndexCloser io.Closer
+	first           bool
+	poolKey         field.Path
 }
 
-type WriterOpts struct {
-	Order order.Which
-	Zng   zngio.WriterOpts
-}
-
-func (r *Reference) NewWriter(ctx context.Context, engine storage.Engine, path *storage.URI, opts WriterOpts) (*Writer, error) {
+func (r *Reference) NewWriter(ctx context.Context, engine storage.Engine, path *storage.URI, o order.Which, seekIndexFactor int, poolKey field.Path) (*Writer, error) {
```

Comment:
Nit: `poolKey` feels more important than `o` or `seekIndexFactor`, so maybe put it first.